src/Ocelot/DependencyInjection/ConfigurationBuilderExtensions.cs

```diff
@@ -28,37 +29,42 @@ public static IConfigurationBuilder AddOcelotBaseUrl(this IConfigurationBuilder
             return builder;
         }
 
-        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder)
+        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, IHostingEnvironment env = null)
         {
-            return builder.AddOcelot(".");
+            return builder.AddOcelot(".", env);
         }
 
-        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, string folder)
+        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, string folder, IHostingEnvironment env = null)
         {
-            const string pattern = "(?i)ocelot\\.([a-zA-Z0-9]*)(\\.json)$";
+            const string primaryConfigFile = "ocelot.json";
 
-            var reg = new Regex(pattern);
+            const string globalConfigFile = "ocelot.global.json";
 
-            var files = Directory.GetFiles(folder)
-                .Where(path => reg.IsMatch(path))
+            const string subConfigPattern = @"^ocelot\.[a-zA-Z0-9]+\.json$";
+
+            string excludeConfigName = env?.EnvironmentName != null ? $"ocelot.{env.EnvironmentName}.json" : string.Empty;
+
+            var reg = new Regex(subConfigPattern, RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+            var files = new DirectoryInfo(folder)
+                .EnumerateFiles()
+                .Where(fi => reg.IsMatch(fi.Name) && (fi.Name != excludeConfigName))
                 .ToList();
 
             var fileConfiguration = new FileConfiguration();
 
             foreach (var file in files)
             {
-                // windows and unix sigh...
-                if(files.Count > 1 && (Path.GetFileName(file) == "ocelot.json"))
+                if(files.Count > 1 && file.Name.Equals(primaryConfigFile, StringComparison.OrdinalIgnoreCase))
```

Comment:
could you not just say and the file isnt ocelot.{env}.json here and then no other code would need to change?